File: Development/demo_unitTest_with_mocks/python/user_service.py
class UserService:
    def __init__(self, external_api):
        self.external_api = external_api

    def get_user_data(self, user_id):
        response = self.external_api.fetch_user(user_id)
        if not response or "name" not in response:
            raise ValueError("Invalid response from external API")
        return f"User: {response['name']}"

File: Development/demo_unitTest_with_mocks/php/UserService.php
<?php
class UserService {
    private $externalApi;

    public function __construct($externalApi) {
        $this->externalApi = $externalApi;
    }

    public function getUserData($userId) {
        $response = $this->externalApi->fetchUser($userId);
        if (!$response || !isset($response['name'])) {
            throw new Exception("Invalid response from external API");
        }
        return "User: " . $response['name'];
    }
}
